### ISSUE TITLE
Game Scanner keeps adding the same game

### ISSUE THREAD

**Issue by mbtns** [moraroy/NonSteamLaunchersDecky#14]
System

> NonSteamLaunchers Version: 1.0.0-6c13f50
> SteamOS Version: 3.5.19
> Steam Deck LCD
> 2 users logged in to the SD, each with NSL plugin active
> Launchers were installed in unique Proton prefix folders

Background

> The NSL plugin was working well until recently, where it did not let me uninstall an added launcher. Even after manually removing the compdata folder of the launcher, the Gamescanner with to Auto Add Games kept adding the non-removed launcher. I was also unable to add any new launchers from this point on, I got a notification that install was completed, but nothing was actually added or installed. I then did a factory reset of the deck and reinstalled everything. The launchers now appear to install again, did not try uninstall functionality.

Main New Issue:

> I installed Diablo IV via the Battle.net launcher, NSL Game Scanner (auto scan enabled) added the game and then proceeded to keep adding the game. I managed to stop this by disabling the Auto Scan Games after 30 library entries were added (had to manually remove these). Tried reinstalling the plugin, but does not help. Also it does not add any new games (tried Alan Wake via GoG) anymore, perhaps these are queued behind the Infinite Diablo's? Not sure how I can resolve this as I am not familiar with how the scanner works. It did not generate a log file in downloads as I guess it does not see this as an "install" activity.

**Comment by moraroy** [comment]
@mbtns not sure what the issue is yet but I would make sure you have the latest decky plugin version, also if you use the desktop version, make sure you have downloaded the latest .desktop file

**Comment by mbtns** [comment]
@moraroy, I use the decky plugin version which appears to be the latest available. I reinstalled the plugin and switched between my 2 accounts and this temporarily appeared to solve the issue, however, when I install a new game (via gog this time), it starts again. I would suspect it has something to do with the plugin being active in 2 accounts on the SD maybe? 

In any case, I guess I will take the manual/Bottles route for now for container management, thanks for the nice work though!

**Comment by moraroy** [comment]
@mbtns oh yeah probably, also not sure if Decky Loader is able to switch between accounts yet so there is also that

**Comment by mbtns** [comment]
> @mbtns oh yeah probably, also not sure if Decky Loader is able to switch between accounts yet so there is also that

Indeed. I did experience decky-loader not loading (hehe) when I switch accounts. Usually need a restart before it is enabled again. So Will create a ticket on the decky-loader git as well.

**Comment by moraroy** [comment]
@mbtns currently in my project NSL, I know that I put the shortcuts in different library accounts for the desktop version, for the decky loader version, I don't think I do just fyi

**Comment by mbtns** [comment]
> @mbtns currently in my project NSL, I know that I put the shortcuts in different library accounts for the desktop version, for the decky loader version, I don't think I do just fyi

Thank you for the info. Will check the desktop version again if I have some time then.
